server/src-lib/Hasura/RQL/DML/RemoteJoin.hs

```diff
@@ -417,7 +416,11 @@ fetchRemoteJoinFields env manager reqHdrs userInfo remoteJoins = do
                                  (map _rjfField batchList)
         gqlReqUnparsed = (GQLQueryText . G.renderExecutableDoc . G.ExecutableDocument . unGQLExecDoc) <$> gqlReq
     -- NOTE: discard remote headers (for now):
-    (_, _, respBody) <- execRemoteGQ' env manager userInfo reqHdrs gqlReqUnparsed rsi G.OperationTypeQuery
+    (_, _, respBody) <- do
+      -- TODO(evertedsphere): move this into metadata, so that a trace further
+      -- down the callstack (i.e. while performing the http req) will print 
+      -- these field aliases.
+      execRemoteGQ' env manager userInfo reqHdrs gqlReqUnparsed rsi G.OperationTypeQuery (toList $ fmap _rjfPath batch)
```

Comment:
I was going to say let's pass a `Maybe (NonEmpty FieldPath)` here, but what about adding something like

```haskell
data RemoteCallReason
  = TopLevelRemoteCall
  -- ^ pass the whole query to the remote schema as an optimization
  | RemoteJoinCall (NonEmpty FieldPath)
  -- ^ fetching a non-empty set of fields in the course of executing a remote join
```

to document the exact nature of the new argument?

Reply:
this sounds like a much better idea